### PR TITLE
git path quoteshell

### PR DESCRIPTION
**Pull request by genotrance**
Fix issue with git execution when in `Program Files`.

cc @timotheecour

### CODE REVIEW

**Comment by genotrance** [comment]
@dom96 PTAL

File: src/choosenimpkg/download.nim [hunk 1]
```diff
@@ -373,7 +373,7 @@ template isDevel*(version: Version): bool =
 
 proc gitUpdate*(version: Version, extractDir: string, params: CliParams): bool =
   if version.isDevel() and params.latest:
-    let git = findExe("git")
+    let git = findExe("git").quoteShell
```
Comment (timotheecour) [inline src/choosenimpkg/download.nim:376]: that's a bug, "".quoteShell.len == 2, not 0

**Comment by timotheecour** [comment]
note: hopefully should fix bug3 and bug4 from https://github.com/dom96/choosenim/issues/192 ; thanks @genotrance for looking into this!

File: src/choosenimpkg/utils.nim [hunk 2]
```diff
@@ -105,7 +105,7 @@ proc getLatestCommit*(repo, branch: string): string =
   if git.len != 0:
     var
       cmd = when defined(windows): "cmd /c " else: ""
-    cmd &= git & " ls-remote " & repo & " " & branch
+    cmd &= git.quoteShell & " ls-remote " & repo & " " & branch
```
Comment (timotheecour) [inline src/choosenimpkg/utils.nim:108]: strformat would be nice here but that's off topic for this PR :)

**Comment by timotheecour** [comment]
after @dom96 approves would be nice to add a git tag so that `choosenim update self` works (IIUC?)

**Comment by genotrance** [comment]
> note: hopefully should fix bug3 and bug4 from #192 ; thanks @genotrance for looking into this!

Can you try out this PR and see if it does?

Presuming you can compile choosenim! I [tested](https://travis-ci.org/github/genotrance/travister/builds/689893114) choosenim last night with the latest `ast2` based nimterop and nimarchive and it built fine on 1.0.6, 1.2.0 and devel across win/lin/osx. It fails on 0.20.2 since choosenim doesn't support that low a version but looks pretty good.

**Comment by timotheecour** [comment]
> Can you try out this PR and see if it does?

how do I do that on windows?
there is no `choosenim update self #head`, and `nimble install choosenim` fails (https://github.com/dom96/choosenim/issues/192#issuecomment-632627101) and so does local cloning of choosenim followed by `nimble install`

>  It fails on 0.20.2 

that's totally fine, making choosenim work for recent nim releases should get priority over making it work for old ones like 0.20.X

**Pull request merged**